src/selectors.js

```diff
@@ -0,0 +1,28 @@
+import { matchPath } from "react-router"
+
+const createSelectors = (structure) => {
+  const { getIn, toJS } = structure
+
+  const getLocation = state => toJS(getIn(state, ['router', 'location']))
+  const getAction = state => toJS(getIn(state, ['router', 'action']))
+  const createMatchSelector = path => {
```

Comment:
Can you add some comments about the behavior of the matchSelector? So, it will be easier to understand for anyone who read the code. Ex. I need to read twice before I realized that `lastPathname` and `lastMatch` are inside a closure and their values are kept in memory, not re-initialized to null everytime.